frame/contracts/src/tests.rs

```diff
@@ -599,6 +604,53 @@ fn migration_on_idle_hooks_works() {
 	}
 }
 
+#[test]
+fn test_ink_compiled_contract() {
+	use serde::Deserialize;
+
+	#[derive(Deserialize)]
+	struct Root {
+		source: Source,
+		spec: Spec,
+	}
+
+	#[derive(Deserialize)]
+	struct Source {
+		wasm: String,
+	}
+
+	#[derive(Deserialize)]
+	struct Spec {
+		constructors: Vec<Constructor>,
+	}
+
+	#[derive(Deserialize)]
+	struct Constructor {
+		selector: String,
+	}
+
+	let contract = std::fs::File::open(fixture_path("flipper.contract")).expect("File not found.");
+	let Root { source, spec } = serde_json::from_reader(contract).expect("Invalid json.");
+	let bytes = hex::decode(source.wasm[2..].to_owned()).expect("Invalid wasm code.");
+
+	let hex_selector = &spec.constructors[0].selector;
+	let selector = u32::from_str_radix(&hex_selector[2..], 16).expect("Invalid selector");
+	let data = selector.to_be_bytes().encode().to_vec();
+
+	ExtBuilder::default().build().execute_with(|| {
+		let _ = Balances::deposit_creating(&ALICE, 100_000_000_000_000);
+		assert_ok!(Contracts::instantiate_with_code(
+			RuntimeOrigin::signed(ALICE),
+			100_000,
```

Comment:
You can't sent balance to an unpayable constructor.

```suggestion
			0,
```

Reply:
🤦